File: bulk_journal_entry/public/js/payment_entry.js
frappe.ui.form.on("Payment Entry", {
	refresh(frm) {
		if (frm.doc.docstatus !== 1 || frm.is_new()) {
			return;
		}

		frm.add_custom_button(
			__("Settlement Journal Entries"),
			() => show_settlement_journal_entries(frm),
			__("View")
		);
	},
});

function show_settlement_journal_entries(frm) {
	frappe.call({
		method: "bulk_journal_entry.api.payment_entry.get_settlement_journal_entries",
		args: {
			payment_entry: frm.doc.name,
		},
		callback(r) {
			const data = r.message || {};
			const rows = data.rows || [];

			if (!rows.length) {
				frappe.msgprint({
					title: __("Settlement Journal Entries"),
					message: __("No submitted Journal Entry rows are linked to this Payment Entry."),
					indicator: "orange",
				});
				return;
			}

			const currency =
				rows[0].account_currency || frm.doc.paid_from_account_currency || frm.doc.paid_to_account_currency;
			const table_rows = rows.map((row) => [
				frappe.utils.get_form_link("Journal Entry", row.journal_entry, true),
				frappe.datetime.str_to_user(row.posting_date),
				row.party || "",
				row.reference_name
					? frappe.utils.get_form_link(row.reference_type, row.reference_name, true)
					: "",
				format_currency(row.settled_amount, row.account_currency || currency),
			]);

			frappe.msgprint({
				title: __("Settlement Journal Entries"),
				message: [
					`<p><b>${__("Total Settled")}:</b> ${format_currency(data.total_settled, currency)}</p>`,
					frappe.render_template("table", {
						data: {
							header: [
								__("Journal Entry"),
								__("Posting Date"),
								__("Party"),
								__("Against Voucher"),
								__("Settled Amount"),
							],
							rows: table_rows,
						},
					}),
				].join(""),
				wide: true,
			});
		},
	});
}

File: bulk_journal_entry/api/payment_entry.py
import frappe


@frappe.whitelist()
def get_settlement_journal_entries(payment_entry):
	if not payment_entry:
		return {"rows": [], "total_settled": 0}

	if not frappe.has_permission("Payment Entry", "read", payment_entry):
		frappe.throw(frappe._("Not permitted"), frappe.PermissionError)

	je_account = frappe.qb.DocType("Journal Entry Account")
	je = frappe.qb.DocType("Journal Entry")

	rows = (
		frappe.qb.from_(je_account)
		.inner_join(je)
		.on(je.name == je_account.parent)
		.select(
			je.name.as_("journal_entry"),
			je.posting_date,
			je.voucher_type,
			je.company,
			je_account.party_type,
			je_account.party,
			je_account.reference_type,
			je_account.reference_name,
			je_account.debit_in_account_currency,
			je_account.credit_in_account_currency,
			je_account.account_currency,
		)
		.where(je.docstatus == 1)
		.where(je_account.docstatus == 1)
		.where(je_account.advance_voucher_type == "Payment Entry")
		.where(je_account.advance_voucher_no == payment_entry)
		.orderby(je.posting_date, je.name)
	).run(as_dict=True)

	total_settled = 0
	for row in rows:
		row.settled_amount = abs(row.debit_in_account_currency or row.credit_in_account_currency or 0)
		total_settled += row.settled_amount

	return {
		"rows": rows,
		"total_settled": total_settled,
	}
